### PR TITLE
VER-3401: cleanup / remove old, unused and obscure code

### PR DESCRIPTION
**Pull request by YuanZHANG910**

### CODE REVIEW

File: app/uk/gov/hmrc/ivorchestration/connectors/AuthConnector.scala [hunk 1]
```diff
@@ -16,14 +16,13 @@
...
+import uk.gov.hmrc.ivorchestration.config.AppConfig
 
+import javax.inject.{Inject, Singleton}
 
-class AuthConnector @Inject()(val http: HttpClient) extends PlayAuthConnector with AuthServiceConfiguration {
-  import authConf._
-
-  lazy val serviceUrl: String = s"$protocol://$host:$port"
+@Singleton
+class AuthConnector @Inject()(val http: HttpClient, app: AppConfig) extends PlayAuthConnector {
```
Comment (stevenmallen) [inline app/uk/gov/hmrc/ivorchestration/connectors/AuthConnector.scala:26]: Just checking - do we still need the auth connector in the backend here?  What is it used for?

File: app/uk/gov/hmrc/ivorchestration/controllers/IvSessionDataController.scala [hunk 2]
```diff
@@ -16,33 +16,30 @@
...
 @Singleton()
 class IvSessionDataController @Inject()(val authConnector: AuthConnector,
                                         headerValidator: HeaderValidator,
                                         ivSessionDataRepository: IvSessionDataRepository,
-                                        cc: ControllerComponents)
+                                        cc: ControllerComponents)(implicit val ec: ExecutionContext)
   extends BackendController(cc) with AuthorisedFunctions with Logging {
 
-  val requestsHandler = new IvSessionDataRequestHandler[Future](ivSessionDataRepository)
+  val requestsHandler = new IvSessionDataRequestHandler(ivSessionDataRepository)
```
Comment (stevenmallen) [inline app/uk/gov/hmrc/ivorchestration/controllers/IvSessionDataController.scala:42]: Nice :)

**Review by stevenmallen — APPROVED** [review]
LGTM - just one question about auth connector

File: app/uk/gov/hmrc/ivorchestration/connectors/AuthConnector.scala (hunk 1 above)
Comment (YuanZHANG910) [reply app/uk/gov/hmrc/ivorchestration/connectors/AuthConnector.scala:26]: Yes, seems like we still need the auth check for the backend.
Clients might call the backend directly, so for safety I think we should keep it.
https://kibana.tools.production.tax.service.gov.uk/app/r/s/HoRi7

**Review by stevenmallen — APPROVED** [review]
LGTM

**Pull request merged**